File: tests/integration/construct_component_unknown/python/component.py
# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

from typing import Optional

import khulnasoft

class Component(khulnasoft.ComponentResource):
    def __init__(self, name: str, args: khulnasoft.Inputs, opts: Optional[khulnasoft.ResourceOptions] = None):
        super().__init__("testcomponent:index:Component", name, args, opts, True)

File: tests/integration/construct_component_error_apply/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as khulnasoft from "@khulnasoft/khulnasoft";

interface ComponentArgs {
    foo: khulnasoft.Input<string>;
}

export class Component extends khulnasoft.ComponentResource {
    public readonly foo!: khulnasoft.Output<string>;

    constructor(name: string, args: ComponentArgs, opts?: khulnasoft.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["foo"] = args.foo;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}

